cg/src/domains/management/subjects/subjects.controller.js

```diff
@@ -1,12 +1,48 @@
 const SubjectsService = require('./subjects.service');
 
+const PAGINATION_COUNT = 2; // HARD CODED CONSTANT
+
 class SubjectsController {
   constructor(subjectsService = null) {
     this.subjectsService = subjectsService || new SubjectsService();
   }
 
   async listSubjects(req, res) {
-    res.json(await this.subjectsService.listSubjects());
+    const subjectsList = await this.subjectsService.listSubjects();
+    var requestedPage = req.query.page;
+    var pages = [];
+
+    if (requestedPage === undefined) {
+      requestedPage = 1;
+    }
+
+    for (var i = 0; i * PAGINATION_COUNT < subjectsList.length; i++) {
```

Comment:
pagination should be done in the DB, take a look at the `limit` and `offset` functions, https://www.petefreitag.com/item/451.cfm